elfloader-tool/src/arch-arm/smp_boot.c

```diff
@@ -117,7 +121,13 @@ WEAK void init_cpus(void)
             abort();
         }
 
-        while (!is_core_up(num_cpus));
+        while (!is_core_up(num_cpus)) {
+#if defined(CONFIG_ARCH_AARCH64)
+            /* The compiler may optimize this loop away, add a dsb()
```

Comment:
Same here, see https://github.com/seL4/seL4_tools/pull/191/commits/31daf6ab48c8f837bbf772d8c5de354b6ced3e0a#r1506631069

Reply:
Please fix the comment here also. I think at this point we should use the same cache settings on all core at least, so the dsb is really enough. And again, removing the `#if defined(CONFIG_ARCH_AARCH64)` and doing this on ARmv7 as well can't hurt.